### ISSUE TITLE
Enhancement: Allow relabel on rings.

### ISSUE THREAD

**Issue by dlnelson** [jmoenig/Snap--Build-Your-Own-Blocks#1699]
I find myself customizing an empty ring with variable names only to discover that I actually need (or would have preferred) a different interior shape.  It would be great if relabel would let me change the interior shape without losing the input variable names.

Unfortunately, this doesn't seem as simple as adding reify* to `blockAlternatives`, because that doesn't preserve the variable names.  `restoreInputs()` needs to be updated to propagate the input names (and possibly also `relabel()` and `setSelector()`).

As I was blindly flapping around, I came up with the below, which fails to preserve the contents of the ring (but worked well enough for me, because I was mostly concerned with empty rings).  Only after did I find that I should have been looking at `restoreInputs()`.  Interestingly, after the relabel blows away the inner content, undo then redo restores it into the new ring.

```javascript
SpriteMorph.prototype.blockAlternatives.reifyScript = ['reifyReporter', 'reifyPredicate'];
SpriteMorph.prototype.blockAlternatives.reifyReporter = ['reifyScript', 'reifyPredicate'];
SpriteMorph.prototype.blockAlternatives.reifyPredicate = ['reifyScript', 'reifyReporter'];

RingMorph.prototype.relabel = function (alternativeSelectors) {
  var menu = new MenuMorph(this),
      oldInputs = this.inputs(),
      oldInputNames = this.inputNames(),     // added
      myself = this;
  alternativeSelectors.forEach(function (sel) {
    var block = SpriteMorph.prototype.blockForSelector(sel),
        slot = block.parts()[1];    // added
    block.restoreInputs(oldInputs);
    oldInputNames.forEach(function (inputName) {     // added
      slot.addInput(inputName);
    });
    block.fixBlockColor(null, true);
    block.addShadow(new Point(3, 3));
    menu.addItem(
      block,
      function() {
        myself.setSelector(sel);
      }
    );
  });
  menu.popup(this.world(), this.bottomLeft().subtract(new Point(
    8,
    this instanceof CommandBlockMorph ? this.corner : 0
  )));
}

RingMorph.prototype.setSelector = function (aSelector) {
    // private - used only for relabel()
    var oldInputs = this.inputs(),
        oldInputNames = this.inputNames(),     // added
        slot,     // added
        scripts = this.parentThatIsA(ScriptsMorph),
        surplus,
        info;
    info = SpriteMorph.prototype.blocks[aSelector];
    this.setCategory(info.category);
    this.selector = aSelector;
    this.setSpec(localize(info.spec));
    surplus = this.restoreInputs(oldInputs);
    slot = this.parts()[1];
    oldInputNames.forEach(function (inputName) {     // added
      slot.addInput(inputName);
    });
    this.fixLabelColor();

    // place surplus blocks on scipts
    if (scripts && surplus.length) {
        surplus.forEach(function (blk) {
            blk.moveBy(10);
            scripts.add(blk);
        });
    }
};
```

**Comment by cycomachead** [comment]
Duplicate of #304
